src/machine/machine_rp2040_adc.go

```diff
@@ -20,43 +39,112 @@ func InitADC() {
 	waitForReady()
 }
 
-// Configure configures a ADC pin to be able to be used to read data.
+// Below methods are for standard tinygo, ADC{} pin based access to the ADC peripheral.
+
+// The Configure method sets the ADC pin to analog input mode.
+// Note the ADCConfig parameter is ignored.
+// Does nothing if the GPIO is not an ADC muxed pin.
 func (a ADC) Configure(config ADCConfig) {
```

Comment:
Why do we have two Configure methods for the same pin (or so it seems from what I can tell)? It seems each pin is configured individually and that each ADC contains a single corresponding Pin? Can't we simplify this to a single `ADCConfig` type which specifies all of the channel configuration- or am I missing something?

Reply:
Yep using ADCConfig makes sense. I was originally thinking the channel config might be quite specific to rp2040 when advanced ADC features get implemented in the future. It's a mute point, Currently no config attributes are used. I'll get rid of the ADCChConfig type.

Reply:
Configuring each individual ADC pin with ADCConfig{} is existing behaviour. Seems a bit odd to me too. The ADC H/W abstraction seems to focus on the input pins rather than a configurable peripheral, with configurable inputs.

Reply:
Maybe having separate ADCs that makes sense? If each input pin can be configured independently it would make sense to have a separate ADC type for each pin. 

I just gave a _very_ cursory glance at the datasheet- it really seems like there is no independent configuring of pins, but I could be wrong.

Reply:
Oh snap, nevermind what I was nagging about then. If that's how other boards do it then so it shall be.

Reply:
OK, so nothing further required here on this point.